rust/libs/logging/src/windows_event_log.rs

```diff
@@ -0,0 +1,456 @@
+//!     .init();
+//! ```
+//!
+//! Event IDs: ERROR=1, WARN=2, INFO/DEBUG/TRACE=3 (supported by `EventCreate.exe`).
+//!
+//! Auto-registers the source on creation (requires admin). To register manually:
+//! ```powershell
+//! New-EventLog -LogName Application -Source "MyApp"
+//! ```
+
+use std::fmt::Write as _;
+use std::sync::{Arc, Mutex};
+
+use tracing::field::{Field, Visit};
+use tracing::span::Attributes;
+use tracing::{Event, Id, Level, Subscriber};
+use tracing_subscriber::filter::Filtered;
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::{EnvFilter, Layer};
+
+use windows::Win32::Foundation::HANDLE;
+use windows::Win32::System::EventLog::{
+    DeregisterEventSource, EVENTLOG_ERROR_TYPE, EVENTLOG_INFORMATION_TYPE, EVENTLOG_WARNING_TYPE,
+    REPORT_EVENT_TYPE, RegisterEventSourceW, ReportEventW,
+};
+use windows::Win32::System::Registry::{
+    HKEY_LOCAL_MACHINE, KEY_WRITE, REG_DWORD, REG_EXPAND_SZ, REG_OPTION_NON_VOLATILE, RegCloseKey,
+    RegCreateKeyExW, RegSetValueExW,
+};
+use windows::core::{PCWSTR, w};
+
+const EVENT_ID_ERROR: u32 = 1;
+const EVENT_ID_WARNING: u32 = 2;
+const EVENT_ID_INFO: u32 = 3;
+const TYPES_SUPPORTED: u32 = 0x07; // Error | Warning | Information
+
+/// Environment variable for controlling Windows Event Log filtering.
+pub const ENV_EVENTLOG_DIRECTIVES: &str = "EVENTLOG_DIRECTIVES";
+const DEFAULT_DIRECTIVES: &str = "info";
+
+/// Thread-safe wrapper around the Windows Event Log handle.
+struct EventSourceHandle {
+    handle: Mutex<Option<HANDLE>>,
+}
+
+// SAFETY: Access is serialized via Mutex.
+unsafe impl Send for EventSourceHandle {}
+unsafe impl Sync for EventSourceHandle {}
+
+impl EventSourceHandle {
+    fn new(handle: HANDLE) -> Self {
+        Self {
+            handle: Mutex::new(Some(handle)),
+        }
+    }
+
+    fn report_event(&self, event_type: REPORT_EVENT_TYPE, event_id: u32, message: &str) {
+        let message_wide = to_wide_string(message);
+        let messages = [PCWSTR(message_wide.as_ptr())];
+
+        let Ok(guard) = self.handle.lock() else {
+            return;
+        };
+        let Some(handle) = *guard else {
+            return;
+        };
+
+        // SAFETY: Handle is valid and protected by mutex.
+        unsafe {
+            let _ = ReportEventW(
+                handle,
+                event_type,
+                0,
+                event_id,
+                None,
+                0,
+                Some(&messages),
+                None,
+            );
+        }
+    }
+}
+
+impl Drop for EventSourceHandle {
+    fn drop(&mut self) {
+        if let Some(handle) = self.handle.get_mut().expect("not poisoned").take() {
+            // SAFETY: We own the handle.
+            unsafe {
+                let _ = DeregisterEventSource(handle);
+            }
+        }
+    }
+}
+
+/// A tracing layer that writes events to the Windows Event Log.
+pub struct WindowsEventLogLayer {
```

Comment:
```suggestion
pub struct Layer {
```

We can refer to it from the module.